features/datachoices/src/main/java/org/opennms/features/datachoices/internal/UsageStatisticsReporter.java

```diff
@@ -362,6 +382,14 @@ public void setProvisiondConfigurationDao(ProvisiondConfigurationDao provisiondC
         m_provisiondConfigurationDao = provisiondConfigurationDao;
     }
 
+    public void setDataSourceFactoryBean(DataSourceFactoryBean dataSourceFactoryBean) {
+        m_dataSourceFactoryBean = dataSourceFactoryBean;
+    }
+
+    public DataSourceFactoryBean getDataSourceFactoryBean() {
```

Comment:
I have a feeling getDataSourceFactoryBean is not e necessary in this scenario. We just need the set method  used by the bluebrint. Otherwise is looks pretty good.!